### PR TITLE
app-shells/starship: add 1.8.0

### PR DESCRIPTION
**Pull request by ixti**
See-Also: https://github.com/gentoo/gentoo/pull/25507

### CODE REVIEW

**Comment by ixti** [comment]
By some reason `pkgdev commit` did not add signed-off-by automatically. Had to rebase. o_O

**Comment by ixti** [comment]
cc: @thesamesam

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
LGTM, except that it does require Rust 1.60 (I confirmed that it does not build with 1.59).

File: app-shells/starship/starship-1.8.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+	ISC
+	MIT
+	MPL-2.0
+	ZLIB
+	WTFPL-2
+"
+SLOT="0"
+KEYWORDS="~amd64"
+
+BDEPEND=">=virtual/rust-1.59"
```
[the remark this answers is not in the document]
Comment (ixti) [reply app-shells/starship/starship-1.8.0.ebuild:334]: I totally overlooked that! Thank you!

**Comment by ixti** [comment]
@bowlofeggs fixed.

**Review by bowlofeggs — APPROVED** [review]
LGTM!

**Comment by bowlofeggs** [comment]
I don't have commit access, so we'll need someone to merge this for us.

**Comment by ixti** [comment]
@bowlofeggs while this PR remained open, a new version was released :D So, I've decided to add it as well, and while at this - drop 1.4.2;

I've used cargo-ebuild to generate a skeleton ebuild, and thus it differs a bit from the previous versions. The main difference is that we will use git archive for the starship, and crates for the dependencies. I've also hand-picked licences from the report of cargo-license. 

In 1.9.0 starship changed one configuration variable, so I've added an info for those upgrading from previous versions.

File: app-shells/starship/starship-1.9.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+pkg_preinst() {
+	UPGRADE_FROM_PRE_1_9_0=0
+
+	if has_version "<${CATEGORY}/${PN}-1.9.0" ; then
+		UPGRADE_FROM_PRE_1_9_0=1
+	fi
+}
+
+pkg_postinst() {
+	if [[ "${UPGRADE_FROM_PRE_1_9_0}" -eq 1 ]] ; then
```
Comment (Flowdalic) [inline app-shells/starship/starship-1.9.1.ebuild:350]: Is there a reason that this is not using `REPLACING_VERSIONS`? That is, simply iterate in `pkg_postinst` over `REPLACING_VERSIONS` and check if there is a version < 1.9.0 using `ver_test`. The you could drop the declaration of `pkg_preinst`.

Besides that, using `0` and `1` is IMHO archaic :) Consider using `true` and `false` instead. For example, this

```bash
foo=0
if something; then
  foo=1
fi

if [[ $foo -eq 1 ]]; then
   do_something
fi
```

could be written more readable as

```bash
foo=false
if something; then
  foo=true
fi

if $foo; then
   do_something
fi
```

**Review by Flowdalic — CHANGES_REQUESTED** [review]

**Comment by Flowdalic** [comment]
If there is no particular reason to add 1.8.0, I''d suggest to drop the commit adding it.

File: app-shells/starship/starship-1.9.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+		einfo "vicmd_symbol config option was renamed to vimcmd_symbol"
```
Comment (Flowdalic) [inline app-shells/starship/starship-1.9.1.ebuild:351]: Please use a full sentence.
```suggestion
		einfo "Note that vicmd_symbol config option was renamed to vimcmd_symbol in version 1.9"
```

File: app-shells/starship/starship-1.9.1.ebuild [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+		einfo " "
```
Comment (Flowdalic) [inline app-shells/starship/starship-1.9.1.ebuild:352]: I'd drop that empty line completely.
```suggestion
```

File: app-shells/starship/starship-1.9.1.ebuild (hunk 2 above)
Comment (ixti) [reply app-shells/starship/starship-1.9.1.ebuild:350]: Thanks for the hint. I was just following [pkg_postinst](https://devmanual.gentoo.org/ebuild-writing/functions/pkg_postinst/index.html) example. But I will use `$REPLACING_VERSIONS`.

Besides that, I find archaic way a bit more readable ;D (and more concise with other tests)

File: app-shells/starship/starship-1.9.1.ebuild (hunk 4 above)
Comment (ixti) [reply app-shells/starship/starship-1.9.1.ebuild:352]: Sure. Followed examples in the doc again ;))

**Comment by ixti** [comment]
@Flowdalic done.

File: app-shells/starship/starship-1.9.1.ebuild [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+	export OPENSSL_NO_VENDOR=true
+
+	cargo_src_configure
+}
+
+src_install() {
+	cargo_src_install
+	dodoc README.md CHANGELOG.md
+
+	einstalldocs
```
Comment (Flowdalic) [inline app-shells/starship/starship-1.9.1.ebuild:338]: I don't see a reason to call `einstalldocs` any more, but maybe I am missing something?

Comment (ixti) [reply app-shells/starship/starship-1.9.1.ebuild:338]: I just copied it blindly. You're right. :D

**Pull request closed without merging**